isis/tests/CMakeLists.txt

```diff
@@ -3,13 +3,13 @@ cmake_minimum_required(VERSION 3.10)
 add_dependencies(isis3 isis3)
 
 file(GLOB test_source "${CMAKE_SOURCE_DIR}/tests/*.cpp")
-
+file(GLOB isis_libs "${CMAKE_BINARY_DIR}/lib/lib*")
```

Comment:
Traditional CMake claims glob=bad but I personally don't care in this case.

Reply:
Good to know! Very little cmake experience here. I didn't take the approach @jessemapel suggested on gitter just because this was easier, but am happy to change it if there are downsides to this.

Reply:
The issue with GLOB is that CMake can't tell when the files change, so it can cause weird issues where you need to re-configure, but CMake doesn't know that. Here it's not a big problem, so I don't think it's an issue.

One think I am concerned about it that this call grabs all of the Plugin libraries too. Here's what I get when I run this on my build:

```
lib/libAdaptiveGruen.dylib		     lib/libJunoCamera.dylib		     lib/libMsiCamera.dylib				lib/libUvvisCamera.dylib
lib/libAlbedo.dylib			     lib/libKaguyaMiCamera.dylib	     lib/libNewHorizonsLeisaCamera.dylib		lib/libVikingCamera.dylib
lib/libAlbedoAtm.dylib			     lib/libKaguyaTcCamera.dylib	     lib/libNewHorizonsLorriCamera.dylib		lib/libVimsCamera.dylib
lib/libAnisotropic1.dylib		     lib/libLambert.dylib		     lib/libNewHorizonsMvicFrameCamera.dylib		lib/libVoyagerCamera.dylib
lib/libAnisotropic2.dylib		     lib/libLambertAzimuthalEqualArea.dylib  lib/libNewHorizonsMvicTdiCamera.dylib		lib/libapollo.dylib
lib/libApolloMetricCamera.dylib		     lib/libLambertConformal.dylib	     lib/libNirCamera.dylib				lib/libcassini.dylib
lib/libApolloPanoramicCamera.dylib	     lib/libLimitPolygonSeeder.dylib	     lib/libNoNormalization.dylib			lib/libchandrayaan1.dylib
lib/libCTXCamera.dylib			     lib/libLoHighCamera.dylib		     lib/libNoOperator.dylib				lib/libclementine.dylib
lib/libChandrayaan1M3Camera.dylib	     lib/libLoMediumCamera.dylib	     lib/libObliqueCylindrical.dylib			lib/libclipper.dylib
lib/libClipperNacRollingShutterCamera.dylib  lib/libLommelSeeliger.dylib	     lib/libOrthographic.dylib				lib/libdawn.dylib
lib/libClipperWacFcCamera.dylib		     lib/libLroNarrowAngleCamera.dylib	     lib/libOsirisRexOcamsCamera.dylib			lib/libgalileo.dylib
lib/libCrismCamera.dylib		     lib/libLroWideAngleCamera.dylib	     lib/libPlanar.dylib				lib/libhayabusa.dylib
lib/libDawnFcCamera.dylib		     lib/libLunarAzimuthalEqualArea.dylib    lib/libPointPerspective.dylib			lib/libhayabusa2.dylib
lib/libDawnVirCamera.dylib		     lib/libLunarLambert.dylib		     lib/libPolarStereographic.dylib			lib/libisis3.dylib
lib/libEquirectangular.dylib		     lib/libLunarLambertEmpirical.dylib      lib/libRingCylindrical.dylib			lib/libjuno.dylib
lib/libForstnerOperator.dylib		     lib/libLunarLambertMcEwen.dylib	     lib/libRobinson.dylib				lib/libkaguya.dylib
lib/libGradientOperator.dylib		     lib/libLwirCamera.dylib		     lib/libRosettaOsirisCamera.dylib			lib/liblo.dylib
lib/libGridPolygonSeeder.dylib		     lib/libMarciCamera.dylib		     lib/libRosettaVirtisCamera.dylib			lib/liblro.dylib
lib/libHapke.dylib			     lib/libMariner10Camera.dylib	     lib/libShade.dylib					lib/libmariner.dylib
lib/libHapkeAtm1.dylib			     lib/libMaximumCorrelation.dylib	     lib/libShadeAtm.dylib				lib/libmer.dylib
lib/libHapkeAtm2.dylib			     lib/libMdisCamera.dylib		     lib/libSimpleCylindrical.dylib			lib/libmessenger.dylib
lib/libHayabusaAmicaCamera.dylib	     lib/libMercator.dylib		     lib/libSinusoidal.dylib				lib/libmex.dylib
lib/libHayabusaNirsCamera.dylib		     lib/libMiniRF.dylib		     lib/libSsiCamera.dylib				lib/libmgs.dylib
lib/libHiresCamera.dylib		     lib/libMinimumDifference.dylib	     lib/libStandardDeviationOperator.dylib		lib/libmro.dylib
lib/libHiriseCamera.dylib		     lib/libMinnaert.dylib		     lib/libStripPolygonSeeder.dylib			lib/libnear.dylib
lib/libHrscCamera.dylib			     lib/libMinnaertEmpirical.dylib	     lib/libTgoCassisCamera.dylib			lib/libnewhorizons.dylib
lib/libHyb2OncCamera.dylib		     lib/libMixed.dylib			     lib/libThemisIrCamera.dylib			lib/libodyssey.dylib
lib/libIdealCamera.dylib		     lib/libMocNarrowAngleCamera.dylib	     lib/libThemisVisCamera.dylib			lib/libosirisrex.dylib
lib/libIsotropic1.dylib			     lib/libMocWideAngleCamera.dylib	     lib/libTopo.dylib					lib/librosetta.dylib
lib/libIsotropic2.dylib			     lib/libMollweide.dylib		     lib/libTopoAtm.dylib				lib/libtgo.dylib
lib/libIssNACamera.dylib		     lib/libMoonAlbedo.dylib		     lib/libTransverseMercator.dylib			lib/libviking.dylib
lib/libIssWACamera.dylib		     lib/libMoravecOperator.dylib	     lib/libUpturnedEllipsoidTransverseAzimuthal.dylib	lib/libvoyager.dylib
```

I don't know if this breaks anything, but ideally we would only link against the lib<module> libraries. All of the plugin libraries are loaded dynamically.

Reply:
Good point on the plugin libraries.